TESTS/mbedmicro-rtos-mbed/rtostimer/main.cpp

```diff
@@ -0,0 +1,398 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include "rtos.h"
+
+#if defined(MBED_RTOS_SINGLE_THREAD)
+#error [NOT_SUPPORTED] test not supported
+#endif
+
+using namespace utest::v1;
+
+#define THREAD_STACK_SIZE 512
+#define TEST_DELAY_MS 50
+#define TEST_DELAY2_MS 30
+#define TEST_DELAY_MS_DELTA 1
+#define TEST_RESTART_DELAY_MS 10
+
+#if TEST_RESTART_DELAY_MS >= TEST_DELAY_MS
+#error invalid TEST_RESTART_DELAY_MS value
+#endif
+
+void timer_callback(void const *arg)
+{
+    Semaphore *sem = (Semaphore *) arg;
+    sem->release();
+}
+
+/** Test one-shot not restarted when elapsed
+ *
+ * Given a one-shot timer
+ * When the timer is started
+ *     and given time elapses
+ * Then the timer stays stopped
+ */
+void test_oneshot_not_restarted()
+{
+    Semaphore sem(1);
+    RtosTimer timer(mbed::callback(timer_callback, (void const *) &sem), osTimerOnce);
+    osStatus stat = timer.stop();
+    TEST_ASSERT_EQUAL(osErrorResource, stat);
+
+    int32_t slots = sem.wait(0);
```

Comment:
Since `Semaphore` has its own tests this assert is actually redundant. I'll fix that.